numba/core/inline_closurecall.py

```diff
@@ -636,12 +647,17 @@ def inline_closure_call(func_ir, glbls, block, i, callee, typingctx=None,
         callee_ir._definitions = ir_utils.build_definitions(callee_ir.blocks)
         numba.core.analysis.dead_branch_prune(callee_ir, arg_typs)
         try:
-            f_typemap, f_return_type, f_calltypes, _ = typed_passes.type_inference_stage(
-                    typingctx, targetctx, callee_ir, arg_typs, None)
-        except Exception as e:
-            f_typemap, f_return_type, f_calltypes, _ = typed_passes.type_inference_stage(
-                    typingctx, targetctx, callee_ir, arg_typs, None)
-            pass
+            pass_result = typed_passes.type_inference_stage(typingctx,
+                                                            targetctx,
+                                                            callee_ir,
+                                                            arg_typs, None)
+        except Exception:
+            pass_result = typed_passes.type_inference_stage(typingctx,
+                                                            targetctx,
+                                                            callee_ir,
+                                                            arg_typs, None)
```

Comment:
Whilst this is clearly a valid refactor, makes me wonder why this is like this!

Reply:
It is rather a strange paradigm... Although not one I intend to understand as part of this change :-)